blob/gcsblob/gcsblob.go

```diff
@@ -134,6 +148,12 @@ const Scheme = "gs"
 //
 //   - access_id: sets Options.GoogleAccessID
 //   - private_key_path: path to read for Options.PrivateKey
+//
+// Currently their use is limited to SignedURL. If left empty they will be guessed
+// which can result in inconsistencies and/or silently fail in environments such as Compute Engine.
+//
+// Absent any other signer and without a private key, access_id will be used
```

Comment:
When would this happen?

IIUC, before this PR, either neither of these fields would be set (using the default URLOpener via lazyCredsOpener, or using URLOpener explicitly but not setting the fields), or they would both be required to be set (only when using URLOpener explicitly).

Your "guessing" code tries to guess both fields. Is there any reason to support only having `access_id` set? I.e., is there any reason to believe that we'll successfully guess one of them but not the other?

If the answer is "no", then I think the "guessing" code is all you need to add; all of the changes below here aren't necessary. Am I missing something?

Reply:
That's for three use cases:
1. On GCE (but not limited to) the private key is not available.
2. The private key has expired (rotation).
3. Tests. (Needs *Service Account Token Creation* privileges.)

Reply:
OK, please add comments to this effect (explaining that in some cases, both fields are available, and in others, only the access_id. Thanks!